src/families/tron/specs.js

```diff
@@ -0,0 +1,152 @@
+        return t;
+      },
+      deviceAction: deviceActions.acceptTransaction,
+      test: ({
+        account,
+        accountBeforeTransaction,
+        transaction,
+        optimisticOperation,
+        operation,
+      }) => {
+        // can be generalized!
+        expect(account.balance.toString()).toBe(
+          accountBeforeTransaction.balance.minus(operation.value).toString()
+        );
+      },
+    },
+    {
+      name: "send max to another account",
+      transaction: ({ account, siblings, bridge }) => {
+        invariant(account.balance.gt(minimalAmount), "balance is too low");
+        let t = bridge.createTransaction(account);
+        const sibling = pickSiblings(siblings);
+        const recipient = sibling.freshAddress;
+        t = bridge.updateTransaction(t, { useAllAmount: true, recipient });
+        return t;
+      },
+      deviceAction: deviceActions.acceptTransaction,
+      test: ({
+        account,
+        accountBeforeTransaction,
+        transaction,
+        optimisticOperation,
+        operation,
+      }) => {
+        expect(account.balance.toString()).toBe("0");
+      },
+    },
+    {
+      name: "freeze 25% to bandwidth | energy",
+      transaction: ({ account, bridge }) => {
+        invariant(account.balance.gt(minimalAmount), "balance is too low");
+        let t = bridge.createTransaction(account);
+
+        const amount = account.balance.div(4).integerValue();
+        t = bridge.updateTransaction(t, {
+          mode: "freeze",
+          amount,
+          resource: account.energy.eq(0) ? "ENERGY" : "BANDWIDTH",
+        });
+        return t;
+      },
+      deviceAction: deviceActions.acceptTransaction,
+      test: ({
+        account,
+        accountBeforeTransaction,
+        transaction,
+        optimisticOperation,
+        operation,
+      }) => {
+        const resourceType = transaction.resource;
+        const expectedAmount = BigNumber(transaction.amount)
+          .times(10e6)
+          .plus(
+            accountBeforeTransaction.tronResources.frozen[`${resourceType}`]
+              ? accountBeforeTransaction.tronResources.frozen[`${resourceType}`]
+                  .amount
+              : BigNumber(0)
+          );
+        expect(expectedAmount.toString()).toBe(
+          account.tronResources.frozen[`${resourceType}`].amount.toString()
+        );
+        const expectedTP = transaction.amount.plus(
+          accountBeforeTransaction.tronResources.tronPower
+        );
+        expect(expectedTP).toBe(account.tronResources.tronPower);
+      },
+    },
+    {
+      name: "unfreeze bandwith / energy",
+      transaction: ({ account, bridge, amount }) => {
+        invariant(account.tronResources.tronPower.gt(0), "no frozen assets");
+        const currentDate = new Date();
+        invariant(
+          account.tronResources.frozen.bandwidth.expiredAt < currentDate &&
+            currentDate > account.tronResources.frozen.energy.expiredAt,
```

Comment:
to be checked what we have in our live-common but i suspect we have helpers function for these that could be at use.